dns/jsonify/main.go

```diff
@@ -0,0 +1,242 @@
+	}
+
+	glog.Infof("benchmarkDirPath: %v\n", benchmarkDirPath)
+	glog.Infof("jsonDirPath: %v\n", jsonDirPath)
+	glog.Infof("benchmarkName: %v\n", benchmarkName)
+
+	latency := perftype.PerfData{Version: "v1"}
+	latencyPerc := perftype.PerfData{Version: "v1"}
+	queries := perftype.PerfData{Version: "v1"}
+	qps := perftype.PerfData{Version: "v1"}
+
+	fileList, err := getFileList(benchmarkDirPath)
+	if err != nil {
+		return fmt.Errorf("listing files error: %v", err)
+	}
+
+	for _, file := range fileList {
+		glog.Infof("processing %v\n", file)
+		result, err := readBenchamrkResult(filepath.Join(benchmarkDirPath, file))
+		if err != nil {
+			return err
+		}
+		latency.DataItems = appendLatency(latency.DataItems, result)
+		latencyPerc.DataItems = appendLatencyPerc(latencyPerc.DataItems, result)
+		queries.DataItems = appendQueries(queries.DataItems, result)
+		qps.DataItems = appendQps(qps.DataItems, result)
+	}
+
+	timeString := time.Now().Format(time.RFC3339)
+	if err = saveMetric(&latency, filepath.Join(jsonDirPath, "Latency_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&latencyPerc, filepath.Join(jsonDirPath, "LatencyPerc_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&queries, filepath.Join(jsonDirPath, "Queries_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&qps, filepath.Join(jsonDirPath, "Qps_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// getFileList returns a list of all files with extension .out.
+func getFileList(dir string) ([]string, error) {
+	var fileNames []string
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return fileNames, err
+	}
+
+	for _, file := range files {
+		if !file.IsDir() && filepath.Ext(file.Name()) == ".out" {
+			fileNames = append(fileNames, file.Name())
+		}
+	}
+	return fileNames, nil
+}
+
+func readBenchamrkResult(path string) (*BenchamrkResult, error) {
+	var result BenchamrkResult
+	bin, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("reading error: %v", err)
+	}
+
+	if err := yaml.Unmarshal(bin, &result); err != nil {
+		return nil, fmt.Errorf("decoding failed: %v", err)
+	}
+	return &result, nil
+}
+
+func toString(v *float64) string {
+	if v == nil {
+		return ""
+	}
+	return fmt.Sprintf("%v", *v)
+}
+
+func createLabels(params *BenchmarkParams) map[string]string {
+	labels := make(map[string]string)
+	labels["run_length_seconds"] = fmt.Sprintf("%v", params.RunLengthSeconds)
+	labels["query_file"] = params.QueryFile
+	labels["kubedns_cpu"] = toString(params.KubednsCpu)
+	labels["dnsmasq_cpu"] = toString(params.DnsmasqCpu)
+	labels["dnsmasq_cache"] = toString(params.DnsmasqCache)
+	labels["max_qps"] = toString(params.MaxQps)
+	return labels
+
+}
+
+func appendLatency(items []perftype.DataItem, result *BenchamrkResult) []perftype.DataItem {
+	labels := createLabels(&result.Params)
```

Comment:
Can we call createLabels once in line 110 and pass on the map to all the append functions?